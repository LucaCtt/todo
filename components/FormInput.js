import React, { memo } from "react";
import { StyleSheet } from "react-native";
import { Input } from "@ui-kitten/components";

const FormInput = ({ style, ...props }) => (
  <Input size="large" {...props} style={[styles.input, style]} />
);

const styles = StyleSheet.create({
  input: {
    marginVertical: 6,
  },
});

export default memo(FormInput);
