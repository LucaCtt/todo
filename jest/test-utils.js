import React from "react";
import { render } from "@testing-library/react-native";
import * as eva from "@eva-design/eva";
import { EvaIconsPack } from "@ui-kitten/eva-icons";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";

const TestProvider = ({ children }) => (
  <>
    <IconRegistry icons={EvaIconsPack} />
    <ApplicationProvider mapping={eva.mapping} theme={eva.light}>
      {children}
    </ApplicationProvider>
  </>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: TestProvider, ...options });

// re-export everything
export * from "@testing-library/react-native";

// override render method
export { customRender as render };
