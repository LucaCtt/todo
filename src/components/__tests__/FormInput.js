import React from "react";
import { render } from "test-utils";

import FormInput from "../FormInput";

describe("<FormInput />", () => {
  test("matches snapshot", () => {
    const { baseElement } = render(<FormInput />);
    expect(baseElement).toMatchSnapshot();
  });
});
