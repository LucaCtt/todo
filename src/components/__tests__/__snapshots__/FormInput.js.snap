// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> matches snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "marginVertical": 6,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 48,
                "paddingHorizontal": 8,
                "paddingVertical": 11,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            appearance="default"
            editable={true}
            placeholderTextColor="#8F9BB3"
            rejectResponderTermination={true}
            size="large"
            status="basic"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                Object {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 4,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
